aws_batch/non_host_alignment.py

```diff
@@ -554,514 +399,455 @@ def wait_for_server_ip(service_name, key_path, remote_username, environment, max
-               aws_batch_job_id, environment, lazy_run = True):
-
-    sample_s3_output_path = sample_s3_output_path.rstrip('/')
-    sample_name = sample_s3_input_path[5:].rstrip('/').replace('/','-')
-    sample_dir = DEST_DIR + '/' + sample_name
-    fastq_dir = sample_dir + '/fastqs'
-    result_dir = sample_dir + '/results'
-    scratch_dir = sample_dir + '/scratch'
-    execute_command("mkdir -p %s %s %s %s" % (sample_dir, fastq_dir, result_dir, scratch_dir))
-    execute_command("mkdir -p %s " % REF_DIR)
-
-    # configure logger
-    log_file = "%s/%s.%s.txt" % (result_dir, LOGS_OUT_BASENAME, aws_batch_job_id)
-    logger = logging.getLogger()
-    logger.setLevel(logging.INFO)
-    handler = logging.FileHandler(log_file)
-    formatter = logging.Formatter("%(asctime)s (%(time_since_last)ss elapsed): %(message)s")
-    handler.addFilter(TimeFilter())
-    handler.setFormatter(formatter)
-    logger.addHandler(handler)
-    # now also echo to stdout so they get to cloudwatch
-    handler = logging.StreamHandler(sys.stdout)
-    handler.setLevel(logging.INFO)
-    formatter = logging.Formatter('(%(time_since_last)ss elapsed): %(message)s')
-    handler.setFormatter(formatter)
-    logger.addHandler(handler)
-    DEFAULT_LOGPARAMS = {"sample_s3_output_path": sample_s3_output_path,
-                         "stats_file": os.path.join(result_dir, STATS_OUT)}
-
-    # Download fastqs
-    command = "aws s3 ls %s/ | grep '\.%s$'" % (sample_s3_input_path, file_type)
-    output = execute_command_with_output(command).rstrip().split("\n")
-    for line in output:
-        m = re.match(".*?([^ ]*." + re.escape(file_type) + ")", line)
-        if m:
-            execute_command("aws s3 cp %s/%s %s/" % (sample_s3_input_path, m.group(1), fastq_dir))
-        else:
-            print "%s doesn't match %s" % (line, file_type)
-
-    fastq_files = execute_command_with_output("ls %s/*.%s" % (fastq_dir, file_type)).rstrip().split("\n")
-
-    if lazy_run:
-       # Download existing data and see what has been done
-        command = "aws s3 cp %s %s --recursive" % (sample_s3_output_path, result_dir)
-        print execute_command_with_output(command)
-
-    # Record total number of input reads
-    initial_file_type_for_log = "fastq_paired" if "fastq" in file_type else "fasta_paired"
-    stats_file = os.path.join(result_dir, STATS_OUT)
-    STATS.append({'total_reads': get_total_initial_reads(fastq_files[0], initial_file_type_for_log, stats_file)})
-
-    # run gsnap remotely
-    if filter_host_flag:
-        gsnapl_input_files = [EXTRACT_UNMAPPED_FROM_SAM_OUT1, EXTRACT_UNMAPPED_FROM_SAM_OUT2]
-        before_file_name_for_log = os.path.join(result_dir, EXTRACT_UNMAPPED_FROM_SAM_OUT1)
-        before_file_type_for_log = "fasta_paired"
+def check_s3_file_presence(s3_path):
+    command = "aws s3 ls %s | wc -l" % s3_path
+    return int(execute_command_with_output(command).rstrip())
```

Comment:
I think this is a bug.  
when you aws s3 ls $non_existence file, it will return 1. you will have to catch it, otherwise the script will fail